### ISSUE TITLE
Error while mysql_install_db on Linux machine but it works on windows normally

### ISSUE THREAD

**Issue by Hassan-Salem** [vorburger/MariaDB4j#333]
I used MariaDB4j for testing and it works locally on windows but it fails on gitlab runner 
Logs
```
"message":"Starting Program [/tmp/MariaDB4j/base/scripts/mysql_install_db, --datadir=/tmp/MariaDB4j/data/3306, --basedir=/tmp/MariaDB4j/base, --no-defaults, --force, --skip-name-resolve] (in working directory /tmp/MariaDB4j/base)","thread_name":"main","errorDetails":""}
405 {"dateTime":"2020-02-17T12:10:24.127Z","component_name":"Api_DTB","level":"INFO","message":"mysql_install_db: /tmp/MariaDB4j/base/scripts/mysql_install_db: line 291: /tmp/MariaDB4j/base/bin/my_print_defaults: not found","thread_name":"Exec Stream Pumper","errorDetails":""}
406 {"dateTime":"2020-02-17T12:10:24.133Z","component_name":"Api_DTB","level":"INFO","message":"mysql_install_db: Installing MariaDB/MySQL system tables in '/tmp/MariaDB4j/data/3306' ...","thread_name":"Exec Stream Pumper","errorDetails":""}
407 {"dateTime":"2020-02-17T12:10:24.134Z","component_name":"Api_DTB","level":"INFO","message":"mysql_install_db: /tmp/MariaDB4j/base/scripts/mysql_install_db: line 506: /tmp/MariaDB4j/base/bin/mysqld: not found","thread_name":"Exec Stream Pumper","errorDetails":""}
408 {"dateTime":"2020-02-17T12:10:24.134Z","component_name":"Api_DTB","level":"INFO","message":"mysql_install_db: ","thread_name":"Exec Stream Pumper","errorDetails":""}
409 {"dateTime":"2020-02-17T12:10:24.134Z","component_name":"Api_DTB","level":"ERROR","message":"mysql_install_db: cat: write error: Broken pipe","thread_name":"Exec Stream Pumper","errorDetails":""}
410 {"dateTime":"2020-02-17T12:10:24.134Z","component_name":"Api_DTB","level":"INFO","message":"mysql_install_db: Installation of system tables failed!  Examine the logs in","thread_name":"Exec Stream Pumper","errorDetails":""}
411 {"dateTime":"2020-02-17T12:10:24.134Z","component_name":"Api_DTB","level":"ERROR","message":"mysql_install_db: cat: write error: Broken pipe","thread_name":"Exec Stream Pumper","errorDetails":""}
412 {"dateTime":"2020-02-17T12:10:24.134Z","component_name":"Api_DTB","level":"INFO","message":"mysql_install_db: /tmp/MariaDB4j/data/3306 for more information.","thread_name":"Exec Stream Pumper","errorDetails":""}
413 {"dateTime":"2020-02-17T12:10:24.136Z","component_name":"Api_DTB","level":"INFO","message":"mysql_install_db: ","thread_name":"Exec Stream Pumper","errorDetails":""}
414 {"dateTime":"2020-02-17T12:10:24.136Z","component_name":"Api_DTB","level":"ERROR","message":"mysql_install_db: cat: write error: Broken pipe","thread_name":"Exec Stream Pumper","errorDetails":""}
415 {"dateTime":"2020-02-17T12:10:24.137Z","component_name":"Api_DTB","level":"INFO","message":"mysql_install_db: The problem could be conflicting information in an external","thread_name":"Exec Stream Pumper","errorDetails":""}
416 {"dateTime":"2020-02-17T12:10:24.137Z","component_name":"Api_DTB","level":"INFO","message":"mysql_install_db: my.cnf files. You can ignore these by doing:","thread_name":"Exec Stream Pumper","errorDetails":""}
417 {"dateTime":"2020-02-17T12:10:24.137Z","component_name":"Api_DTB","level":"INFO","message":"mysql_install_db: ","thread_name":"Exec Stream Pumper","errorDetails":""}
418 {"dateTime":"2020-02-17T12:10:24.137Z","component_name":"Api_DTB","level":"INFO","message":"mysql_install_db:     shell> /tmp/MariaDB4j/base/scripts/mysql_install_db --defaults-file=~/.my.cnf","thread_name":"Exec Stream Pumper","errorDetails":""}
419 {"dateTime":"2020-02-17T12:10:24.138Z","component_name":"Api_DTB","level":"INFO","message":"mysql_install_db: ","thread_name":"Exec Stream Pumper","errorDetails":""}
420 {"dateTime":"2020-02-17T12:10:24.138Z","component_name":"Api_DTB","level":"INFO","message":"mysql_install_db: You can also try to start the mysqld daemon with:","thread_name":"Exec Stream Pumper","errorDetails":""}
421 {"dateTime":"2020-02-17T12:10:24.138Z","component_name":"Api_DTB","level":"INFO","message":"mysql_install_db: ","thread_name":"Exec Stream Pumper","errorDetails":""}
422 {"dateTime":"2020-02-17T12:10:24.138Z","component_name":"Api_DTB","level":"INFO","message":"mysql_install_db:     shell> /tmp/MariaDB4j/base/bin/mysqld --skip-grant --general-log &","thread_name":"Exec Stream Pumper","errorDetails":""}
423 {"dateTime":"2020-02-17T12:10:24.138Z","component_name":"Api_DTB","level":"INFO","message":"mysql_install_db: ","thread_name":"Exec Stream Pumper","errorDetails":""}
424 {"dateTime":"2020-02-17T12:10:24.139Z","component_name":"Api_DTB","level":"INFO","message":"mysql_install_db: and use the command line tool /tmp/MariaDB4j/base/bin/mysql","thread_name":"Exec Stream Pumper","errorDetails":""}
425 {"dateTime":"2020-02-17T12:10:24.139Z","component_name":"Api_DTB","level":"INFO","message":"mysql_install_db: to connect to the mysql database and look at the grant tables:","thread_name":"Exec Stream Pumper","errorDetails":""}
426 {"dateTime":"2020-02-17T12:10:24.139Z","component_name":"Api_DTB","level":"INFO","message":"mysql_install_db: ","thread_name":"Exec Stream Pumper","errorDetails":""}
427 {"dateTime":"2020-02-17T12:10:24.139Z","component_name":"Api_DTB","level":"INFO","message":"mysql_install_db:     shell> /tmp/MariaDB4j/base/bin/mysql -u root mysql","thread_name":"Exec Stream Pumper","errorDetails":""}
428 {"dateTime":"2020-02-17T12:10:24.139Z","component_name":"Api_DTB","level":"INFO","message":"mysql_install_db:     mysql> show tables;","thread_name":"Exec Stream Pumper","errorDetails":""}
429 {"dateTime":"2020-02-17T12:10:24.140Z","component_name":"Api_DTB","level":"INFO","message":"mysql_install_db: ","thread_name":"Exec Stream Pumper","errorDetails":""}
430 {"dateTime":"2020-02-17T12:10:24.140Z","component_name":"Api_DTB","level":"INFO","message":"mysql_install_db: Try 'mysqld --help' if you have problems with paths.  Using","thread_name":"Exec Stream Pumper","errorDetails":""}
431 {"dateTime":"2020-02-17T12:10:24.140Z","component_name":"Api_DTB","level":"INFO","message":"mysql_install_db: --general-log gives you a log in /tmp/MariaDB4j/data/3306 that may be helpful.","thread_name":"Exec Stream Pumper","errorDetails":""}
432 {"dateTime":"2020-02-17T12:10:24.140Z","component_name":"Api_DTB","level":"INFO","message":"mysql_install_db: ","thread_name":"Exec Stream Pumper","errorDetails":""}
433 {"dateTime":"2020-02-17T12:10:24.140Z","component_name":"Api_DTB","level":"INFO","message":"mysql_install_db: The latest information about mysql_install_db is available at","thread_name":"Exec Stream Pumper","errorDetails":""}
434 {"dateTime":"2020-02-17T12:10:24.141Z","component_name":"Api_DTB","level":"INFO","message":"mysql_install_db: https://mariadb.com/kb/en/installing-system-tables-mysql_install_db","thread_name":"Exec Stream Pumper","errorDetails":""}
435 {"dateTime":"2020-02-17T12:10:24.141Z","component_name":"Api_DTB","level":"INFO","message":"mysql_install_db: MariaDB is hosted on launchpad; You can find the latest source and","thread_name":"Exec Stream Pumper","errorDetails":""}
436 {"dateTime":"2020-02-17T12:10:24.141Z","component_name":"Api_DTB","level":"INFO","message":"mysql_install_db: email lists at http://launchpad.net/maria","thread_name":"Exec Stream Pumper","errorDetails":""}
437 {"dateTime":"2020-02-17T12:10:24.141Z","component_name":"Api_DTB","level":"INFO","message":"mysql_install_db: ","thread_name":"Exec Stream Pumper","errorDetails":""}
438 {"dateTime":"2020-02-17T12:10:24.141Z","component_name":"Api_DTB","level":"INFO","message":"mysql_install_db: Please check all of the above before submitting a bug report","thread_name":"Exec Stream Pumper","errorDetails":""}
439 {"dateTime":"2020-02-17T12:10:24.142Z","component_name":"Api_DTB","level":"INFO","message":"mysql_install_db: at http://mariadb.org/jira","thread_name":"Exec Stream Pumper","errorDetails":""}
440 {"dateTime":"2020-02-17T12:10:24.142Z","component_name":"Api_DTB","level":"INFO","message":"mysql_install_db: ","thread_name":"Exec Stream Pumper","errorDetails":""}
441 {"dateTime":"2020-02-17T12:10:24.143Z","component_name":"Api_DTB","level":"ERROR","message":"Program [/tmp/MariaDB4j/base/scripts/mysql_install_db, --datadir=/tmp/MariaDB4j/data/3306, --basedir=/tmp/MariaDB4j/base, --no-defaults, --force, --skip-name-resolve] (in working directory /tmp/MariaDB4j/base) failed unexpectedly","thread_name":"Exec Default Executor","errorDetails":"org.apache.commons.exec.ExecuteException: Process exited with an error: 1 (Exit value: 1)\n\tat org.apache.commons.exec.DefaultExecutor.executeInternal(DefaultExecutor.java:404)\n\tat org.apache.commons.exec.DefaultExecutor.access$200(DefaultExecutor.java:48)\n\tat org.apache.commons.exec.DefaultExecutor$1.run(DefaultExecutor.java:200)\n\tat java.lang.Thread.run(Thread.java:745)\n"}
442 {"dateTime":"2020-02-17T12:10:24.220Z","component_name":"Api_DTB","level":"ERROR","x-iosw-wsse":"UsernameToken Username=\"WKF-WKF\"","ftapplicationroles":"l3-teaming l3-connectpro CTY-01 ADMINOBS_RW","session_id":"1581941417404306","sm_universalid":"DTBS0001","message":"Program [/tmp/MariaDB4j/base/scripts/mysql_install_db, --datadir=/tmp/MariaDB4j/data/3306, --basedir=/tmp/MariaDB4j/base, --no-defaults, --force, --skip-name-resolve] (in working directory /tmp/MariaDB4j/base) failed","thread_name":"main","errorDetails":""}
443 105218 [ERROR] Tests run: 1, Failures: 0, Errors: 1, Skipped: 0, Time elapsed: 6.674 s
```
version i used
```
<dependency>
            <groupId>ch.vorburger.mariaDB4j</groupId>
            <artifactId>mariaDB4j</artifactId>
            <version>2.4.0</version>
        </dependency>
```
code to start db

```
DB = new MariaDB4jSpringService();
DB.setDefaultPort(3306);
DB.start();
DB.getDB().createDB("configuration");
```

**Comment by vorburger** [comment]
This may be something related to a more restricted environment on GitLab
runner?

The log you posted above is kind of hard to "parse" in 30s on a mobile for
me.. but have you had a closer look at it yourself if anything stands out
to explain why the process fails to start?

>

**Comment by Mawster** [comment]
Hello,

had the same problem. I'm using gitlab-ci building my project using a openjdk:8 image and I also got the problem that the startup failed (on windows everything worked fine). 

In one of my log messages I found this line: error while loading shared libraries: libncurses.so.5
I fixxed my problem with a before script:
```  
- apt update
- apt-get --assume-yes install libncurses5
```

Maybe this helps someone coming into this thread.
